dspace-rest/src/main/java/org/dspace/rest/filter/ItemFilterUtil.java

```diff
@@ -0,0 +1,263 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+ package org.dspace.rest.filter;
+
+import java.sql.SQLException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.apache.log4j.Logger;
+import org.dspace.content.Bitstream;
+import org.dspace.content.Bundle;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.ConfigurationManager;
+import org.dspace.core.Context;
+
+import com.ibm.icu.util.Calendar;
+
+public class ItemFilterUtil {
+    protected static ItemService itemService = ContentServiceFactory.getInstance().getItemService();
+    static Logger log = Logger.getLogger(ItemFilterUtil.class);
+	public enum BundleName{ORIGINAL,TEXT,LICENSE,THUMBNAIL;}
+	
+	static String getDocumentMimeTypes() {
+		return ConfigurationManager.getProperty("rest", "rest-report-mime-document");
+	}
+	
+	static String getSupportedDocumentMimeTypes() {
+		return ConfigurationManager.getProperty("rest", "rest-report-mime-document-supported");
+	}
+
+	static String getSupportedImageMimeTypes() {
+		return ConfigurationManager.getProperty("rest", "rest-report-mime-document-image");
+	}
+
+	static int countOriginalBitstream(Item item) {
+		return countBitstream(BundleName.ORIGINAL, item);
+	}
+	static int countBitstream(BundleName bundleName, Item item) {
+		int count = 0;
+        for(Bundle bundle: item.getBundles()){
+        	if (!bundle.getName().equals(bundleName.name())){
+        		continue;
+        	}
+        	count += bundle.getBitstreams().size();
+        }
+
+		return count;
+	}
+
+	static List<String> getBitstreamNames(BundleName bundleName, Item item) {
+		ArrayList<String> names = new ArrayList<String>();
+        for(Bundle bundle: item.getBundles()){
+        	if (!bundle.getName().equals(bundleName.name())){
+        		continue;
+        	}
+        	for(Bitstream bit: bundle.getBitstreams()) {
+        		names.add(bit.getName());
+        	}
+        }
+		return names;
+	}
+
+	
+	static int countOriginalBitstreamMime(Context context, Item item, String mimeList) {
+		return countBitstreamMime(context, BundleName.ORIGINAL, item, mimeList);
+	}
+	static int countBitstreamMime(Context context, BundleName bundleName, Item item, String mimeList) {
+		int count = 0;
+        for(Bundle bundle: item.getBundles()){
+        	if (!bundle.getName().equals(bundleName.name())){
+        		continue;
+        	}
+        	for(Bitstream bit: bundle.getBitstreams()) {
+            	for(String mime: mimeList.split(",")) {
+                    try {
+						if (bit.getFormat(context).getMIMEType().equals(mime)) {
+							count++;
```

Comment:
Perhaps also trim the mime ? Because when an space occurs between the commas the code will not work properly.